### ISSUE TITLE
[Question] defining changes to arbitrary properties

### ISSUE THREAD

**Issue by futuun** [software-mansion/react-native-reanimated#140]
**TLDR:** I’m looking for solution to define how arbitrary properties (arrays, strings, objects etc) should react to changes in animated value.
##  

I’ve been playing with reanimated for few months and I refactored quite few complex animated components. Everything seems great up until you want to react to interaction with something else than numbers.
Let say that you have animated value that goes from 0 to 2 and you want to round it and map to three wise monkeys [‘🙈’, ‘🙉’, ‘🙊’] and pass one of them as props and update it every time animated value changes. I could think of few solutions:

### 1. Nested cond nodes

```js
A.cond(A.eq(interpolation, 0),
    '🙈',
    A.cond(A.eq(interpolation, 1),
        '🙉',
        A.cond(A.eq(interpolation, 2),
            '🙊',
        ),
    ),
)
```

### 2. Call node

You could use call node to get current value and use `setNativeProps` from js thread and update text input to whatever you want. The whole point of this lib is to declaratively say what should happened and not go back and forth between js and native.

### 3. Macro

They are fun to write you can pick any syntax you like. Under the hood it'll create tons of nested cond nodes but it should be easier to maintain and in most cases you shouldn’t see performance issues.

### 4. Custom text component

You could write component that takes unicode and shows proper character. It could work for monkeys, they have consecutive codes [0x1F648, 0x1F649, 0x1F64A] but with custom mix of characters  it would easily become unmaintainable.

None of those 4 seemed appealing enough so I decided that I need new kind of node.
It has pretty straight forward syntax: `index([‘🙈’, ‘🙉’, ‘🙊’], animatedValue)` and [It works](https://github.com/futuun/react-native-reanimated/commit/e93091b420151f2c062a702d0abcc8a7266f50fd) for my use case.

Is that more common problem? I'm not sure how other peoples are using that and if I'm not trying to use it as silver bullet 👨🔫

**Comment by osdnk** [comment]
Hi @futuun 
Frankly speaking we was not focusing on other features that animations and style-related issues. The motivation of this library is to optimise complex animation because it was an issue in React Native, which was calling for such solution. I do not have a strong opinion, but I rather think that there's no strong need for optimising operations you mentioned, but I'm going to discuss it with @kmagiera.

**Comment by kmagiera** [comment]
Closing this as we won't be adding big new features to reanimated 1. This kind of functionality is covered in the new Reanimated 2 and we would rather focus on stabilizing the new API and hope y'all will try it soon and that it will resolve the need of having such function in Reanimated 1.x
